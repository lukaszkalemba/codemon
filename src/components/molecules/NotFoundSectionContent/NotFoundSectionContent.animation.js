const getButton = isMobile => ({
  variants: {
    initial: {
      opacity: 0,
      x: isMobile ? -30 : -80,
    },
    animate: {
      opacity: 1,
      x: 0,
      transition: {
        duration: 0.75,
        delay: 0.25,
      },
    },
  },
  initial: 'initial',
  animate: 'animate',
});

export default {
  getButton,
};
