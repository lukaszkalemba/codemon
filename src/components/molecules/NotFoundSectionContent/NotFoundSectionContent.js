import React from 'react';
import useWindowSize from 'hooks/useWindowSize';
import Button from 'components/atoms/Button';
import Container from 'components/particles/Container';
import animations from './NotFoundSectionContent.animation';
import S from './NotFoundSectionContent.styles';

const NotFoundSectionContent = () => {
  const size = useWindowSize();

  const isMobile = size.width < 1150;

  const buttonAnimation = animations.getButton(isMobile);

  return (
    <Container axis="both">
      <S.Heading tag="h1">Coś poszło nie tak...</S.Heading>
      <S.ButtonWrapper {...buttonAnimation}>
        <Button type="link" path="/" priority="secondary">
          Strona główna
        </Button>
      </S.ButtonWrapper>
    </Container>
  );
};

export default NotFoundSectionContent;
