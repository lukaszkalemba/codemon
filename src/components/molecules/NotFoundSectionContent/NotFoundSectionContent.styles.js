import styled from 'styled-components';
import { motion } from 'framer-motion';
import HeadingComponent from 'components/atoms/Heading';

const ButtonWrapper = styled(motion.div)`
  width: 50%;
  display: flex;
  justify-content: flex-start;
  align-items: center;
  margin-left: 35px;

    ${({ theme }) => theme.media.tablet`
    margin-left: 75px;
  `}

  ${({ theme }) => theme.media.laptop`
    justify-content: center;
    margin-left: -25px;
  `}

  ${({ theme }) => theme.media.desktop`
    justify-content: center;
    margin-left: 40px;
  `}
`;

const Heading = styled(HeadingComponent)`
  margin: 1.5em 0 1em;

  ${({ theme }) => theme.media.laptop`
    font-size: ${theme.font.size.heading.l};
    margin-bottom: 0.55em;
  `}

  ${({ theme }) => theme.media.desktop`
    font-size: ${theme.font.size.heading.xl};
  `}
`;

export default { ButtonWrapper, Heading };
