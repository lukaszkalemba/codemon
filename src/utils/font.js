export default {
  size: {
    xxxs: '1rem',
    xxs: '1.3rem',
    xs: '1.6rem',
    s: '2.2rem',
    m: '3rem',
    l: '4rem',
    xl: '5.4rem',
    xxl: '8.5rem',
    xxxl: '10rem',
  },

  weight: {
    regular: '400',
    medium: '500',
    bold: '700',
  },

  family: {
    montserrat: 'Montserrat',
  },
};
