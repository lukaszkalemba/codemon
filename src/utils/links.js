export default [
  {
    id: 0,
    path: '/projekty',
    name: 'projekty',
  },
  {
    id: 1,
    path: '/blog',
    name: 'blog',
  },
  {
    id: 2,
    path: '/wycena',
    name: 'wycena',
  },
];
