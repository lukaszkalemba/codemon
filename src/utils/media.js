export default {
  phone: '576px',
  tablet: '768px',
  desktop: '1150px',
  largeDesktop: '1600px',
};
