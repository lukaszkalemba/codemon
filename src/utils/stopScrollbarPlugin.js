import { ScrollbarPlugin } from 'smooth-scrollbar';

class stopScrollbarPlugin extends ScrollbarPlugin {
  static pluginName = 'stopScrollbar';

  static defaultOptions = {
    open: false,
  };

  transformDelta(delta) {
    return this.options.open ? { x: 0, y: 0 } : delta;
  }
}

export default stopScrollbarPlugin;
