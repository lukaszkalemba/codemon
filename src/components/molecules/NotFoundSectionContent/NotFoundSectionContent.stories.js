import React from 'react';
import NotFoundSectionContent from './NotFoundSectionContent';

export const notFoundSectionContent = () => <NotFoundSectionContent />;

export default {
  component: NotFoundSectionContent,
  title: 'Molecules|NotFoundSectionContent',
};
