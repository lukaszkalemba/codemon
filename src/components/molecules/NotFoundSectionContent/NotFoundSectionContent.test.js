import React from 'react';
import { render } from 'utils/test-utils';
import '__mocks__/intersectionObserver';
import NotFoundSectionContent from './NotFoundSectionContent';

describe('<NotFoundSectionContent />', () => {
  it('renders correctly and matches snapshot', () => {
    const { container, getByRole } = render(<NotFoundSectionContent />);

    const heading = getByRole('heading');
    const button = getByRole('button');

    expect(heading).toBeInTheDocument();
    expect(button).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
