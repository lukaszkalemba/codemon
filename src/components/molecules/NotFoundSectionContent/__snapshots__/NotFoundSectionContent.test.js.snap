// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFoundSectionContent /> renders correctly and matches snapshot 1`] = `
<main>
  <div
    class="sc-AxiKw knHbTu"
  >
    <h1
      class="sc-AxhCb jcEHSw sc-AxheI biJvfv"
      style="opacity: 0; transform: translateX(40px) translateZ(0);"
    >
      Coś poszło nie tak...
    </h1>
    <div
      class="sc-AxgMl fOVlnj"
    >
      <a
        class="sc-AxjAm bgfZmU"
        href="/"
      >
        <button
          class="sc-AxirZ cvvsKc"
          type="button"
        >
          <span>
            Strona główna
          </span>
        </button>
      </a>
    </div>
  </div>
</main>
`;
